src/services/mongoose/users.ts

```diff
@@ -48,21 +45,11 @@ export const UsersRepository = (): IUsersRepository => {
         deviceToken: deviceTokens,
         lastConnection,
       }
-      const doc = await User.updateOne({ _id: id }, { $set: data })
-      if (doc.nModified !== 1) {
+      const user: User = await User.findOneAndUpdate({ _id: id }, data)
```

Comment:
I just noticed I missed this change, but would casting the return from `findOneAndUpdate` like this instead of explicitly defining the props bring in extra props that we don't necessarily want to have on an object of `User` type? (cc @samerbuna, @bodymindarts)

Reply:
Fixed in #579